server/utils.go

```diff
@@ -91,6 +100,39 @@ func SendHTTPRequest(ctx context.Context, client http.Client, method, url string
 	return resp.StatusCode, nil
 }
 
+// SendHTTPRequestWithRetries - prepare and send HTTP request, retrying the request if within the client timeout
+func SendHTTPRequestWithRetries(ctx context.Context, client http.Client, method, url string, userAgent UserAgent, payload, dst any, handleError ErrorHandlerFunc) (code int, err error) {
```

Comment:
it may be simpler to pass in the logger instance, and log from inside, rather than pass in an error handler?